### ISSUE TITLE
Can't run jekyll serve

### ISSUE THREAD

**Issue by sabgaby** [DIYbiosphere/sphere.dir#58]
I can't run internally the build system. It tells me I am missing gems but they are installed, I can even check the version. However, I can do `bundle exec jekyll install` and it will run. Here is a screenshot
![screen shot 2016-06-03 at 10 18 44](https://cloud.githubusercontent.com/assets/5911330/15772997/d2749c64-2974-11e6-942a-b63928a632a1.png)

**Comment by RaoOfPhysics** [comment]
Testing this.

**Comment by RaoOfPhysics** [comment]
Even more issues at my end. :P

```bash
usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/resolver.rb:356:in `block in verify_gemfile_dependencies_are_found!': Could not find gem 'sanitize (~> 4.0.1)' in any of the gem sources listed in your Gemfile or available on this machine. (Bundler::GemNotFound)
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/resolver.rb:331:in `each'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/resolver.rb:331:in `verify_gemfile_dependencies_are_found!'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/resolver.rb:200:in `start'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/resolver.rb:184:in `resolve'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/definition.rb:200:in `resolve'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/definition.rb:140:in `specs'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/definition.rb:185:in `specs_for'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/definition.rb:174:in `requested_specs'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/environment.rb:19:in `requested_specs'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:14:in `setup'
	from /usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
	from /usr/local/lib/ruby/gems/2.3.0/gems/jekyll-3.1.6/lib/jekyll/plugin_manager.rb:33:in `require_from_bundler'
	from /usr/local/lib/ruby/gems/2.3.0/gems/jekyll-3.1.6/bin/jekyll:9:in `<top (required)>'
	from /usr/local/bin/jekyll:23:in `load'
	from /usr/local/bin/jekyll:23:in `<main>'
```

Installing relevant gems.

**Comment by RaoOfPhysics** [comment]
FWIW, it doesn't seem like the problem is exclusive to `jemoji`. I replaced it with `github-pages` in `_config.yml` and it is giving me an identical error claiming `github-pages` isn't installed. Troubleshooting…

**Comment by sabgaby** [comment]
I know.. it's like this since Wednesday. The problem at first was `gem jekyll-last-modified-at`, and then it was something else, and then `jemoji`.. I don't know what's going on. For now I am bypassing the issue with the `exec serve` command.. Since I don't know programming this is me right now. 
![post-64231-this-is-fine-dog-fire-comic-im-n7mp](https://cloud.githubusercontent.com/assets/5911330/15773691/a4da7db0-2978-11e6-8ccf-0dc34284eaad.png)

**Comment by RaoOfPhysics** [comment]
It may be that that's the correct way forwards: https://stackoverflow.com/questions/35167549/jekyll-dependency-error-when-upgrading-from-2-x-to-3-x

I'm still investigating.

**Comment by blahah** [comment]
So, the best way to get up and running is to use bundler. TLDR at the bottom...

Ruby packages (gems) are generally shared stored in a central location. But this can lead to conflicts between versions required by different things, so Bundler was invented as a way to keep a nice tidy local set of gems installed for a particular project. If you do `jekyll serve` it looks in the central install of gems, and the ones that your jekyll setup needs are probably not installed there.

Instead you run `bundle exec <command>` which takes everything after `bundle exec` and runs it in the context of the nearest local gem store (usually the current directory). So `bundle exec jekyll serve` is the best way to get the server running. And `bundle install` is the best way to make sure everything is installed in the local directory.

### TLDR

```
bundle install
bundle exec jekyll serve
```

**Comment by RaoOfPhysics** [comment]
Lesson learnt: "When in doubt, give @blahah a shout."

**Comment by blahah** [comment]
`s/shout/food reward/` 👌

**Comment by sabgaby** [comment]
@blahah and why do I also get this: 
`GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.`

**Comment by RaoOfPhysics** [comment]
Doesn't rhyme. No can do. ;)

**Comment by blahah** [comment]
@sabgaby that's because the [`github-metadata`](https://github.com/jekyll/github-metadata) jekyll plugin is installed, but no Github API key has been provided.

I don't think the plugin is actually being used, so you can remove it from the `_config.yml` and the message should stop...

**Comment by blahah** [comment]
@RaoOfPhysics `s/shout/stout/` 🍺  ?

**Comment by RaoOfPhysics** [comment]
`s/shout/stoat/`

![](http://www.factzoo.com/sites/all/img/mammals/weasel/stoat-grass.jpg)

Sorry about clogging your issue, @sabgaby. :P

**Comment by blahah** [comment]
stoat doesn't rhyme with shout....

**Comment by RaoOfPhysics** [comment]
Not with that attitude…

**Comment by sabgaby** [comment]
:tada: Solution!:
1. In the terminal `cd` to the repository
2. Run `gem cleanup` 
3. Run `gem list`, check that "jekyll" is there.
4. Run `sudo bundle clean --force` and input your password
5. Run `jekyll serve`

This should work!

**Comment by RaoOfPhysics** [comment]
Can confirm, it works! :D
